src/tess_sip/tess_sip.py

```diff
@@ -92,65 +99,101 @@ def SIP(
             period_at_max_power: the best fit period of the sinusoid.
             power_bkg: the power at each period for the pixels -outside- the aperture
             raw_lc_bkg: the background light curve (pixels outside aperture)
+            model: the systematics model used to correct the light curve
     """
 
-    # Get the un-background subtracted data
-    if hasattr(tpfs[0], "flux_bkg"):
-        tpfs_uncorr = [
-            (tpf + np.nan_to_num(tpf.flux_bkg))[
-                np.isfinite(np.nansum(tpf.flux_bkg, axis=(1, 2)))
+    if ((type(tpfs) is lk.collections.TargetPixelFileCollection) or 
+            (type(tpfs) is lk.targetpixelfile.TessTargetPixelFile)):
+        # Get the un-background subtracted data
+        print("Target Pixel File Input")
+        if hasattr(tpfs[0], "flux_bkg"):
+            tpfs_uncorr = [
+                (tpf + np.nan_to_num(tpf.flux_bkg.value))[
+                    np.isfinite(np.nansum(tpf.flux_bkg.value, axis=(1, 2)))
+                ]
+                for tpf in tpfs
             ]
-            for tpf in tpfs
-        ]
-    else:
-        tpfs_uncorr = tpfs
+        else:
+            tpfs_uncorr = tpfs
 
-    apers = [
-        tpf.pipeline_mask
-        if tpf.pipeline_mask.any()
-        else tpf.create_threshold_mask(aperture_threshold)
-        for tpf in tpfs_uncorr
-    ]
-    bkg_apers = [
-        (~aper) & (np.nansum(tpf.flux, axis=0) != 0)
-        for aper, tpf in zip(apers, tpfs_uncorr)
-    ]
-    lc = (
-        lk.LightCurveCollection(
-            [
-                tpf.to_lightcurve(aperture_mask=aper)
-                for tpf, aper in zip(tpfs_uncorr, apers)
-            ]
+        apers = [
```

Comment:
I think some of these sections have perhaps not been linted quite right, make sure to run `flake8` on these so they lint properly?

Reply:
I checked linting using `flake8` and updated accordingly.